tests/unit/test_auth_proxy.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Jenkins-k8s auth_proxy unit tests."""
+
+# pylint:disable=protected-access
+
+from unittest.mock import ANY, MagicMock, patch
+
+import ops
+from charms.oathkeeper.v0.auth_proxy import AuthProxyRequirer
+from charms.traefik_k8s.v2.ingress import IngressPerAppRequirer
+from ops.testing import Harness
+
+import auth_proxy
+import ingress
+from charm import INGRESS_RELATION_NAME, JenkinsK8sOperatorCharm
+
+CHARM_METADATA = """
+name: test-charm
+requires:
+  auth-proxy:
+    interface: auth_proxy
+  ingress:
+    interface: ingress
+containers:
+  jenkins:
+    resource: jenkins-image
+"""
+
+
+class TestCharm(ops.CharmBase):
```

Comment:
Could anycharm be used here?

Reply:
It seems like a bit overkill since this is a unit tests and we don't really need the other side of the relation. This approach is also used in some other charms for unit testing libraries